src/server/scripts/Spells/spell_holiday.cpp

```diff
@@ -277,6 +277,110 @@ class spell_hallow_end_tricky_treat : public SpellScriptLoader
         }
 };
 
+// Hallowed wands
+enum HallowendData
+{
+    //wand spells
+    SPELL_HALLOWED_WAND_PIRATE             = 24717,
+    SPELL_HALLOWED_WAND_NINJA              = 24718,
+    SPELL_HALLOWED_WAND_LEPER_GNOME        = 24719,
+    SPELL_HALLOWED_WAND_RANDOM             = 24720,
+    SPELL_HALLOWED_WAND_SKELETON           = 24724,
+    SPELL_HALLOWED_WAND_WISP               = 24733,
+    SPELL_HALLOWED_WAND_GHOST              = 24737,
+    SPELL_HALLOWED_WAND_BAT                = 24741
+};
+
+class spell_hallow_end_wand : public SpellScriptLoader
+{
+public:
+    spell_hallow_end_wand() : SpellScriptLoader("spell_hallow_end_wand") {}
+
+    class spell_hallow_end_wand_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_hallow_end_wand_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellEntry*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_PIRATE_COSTUME_MALE))
+                return false;
+            if (!sSpellMgr->GetSpellInfo(SPELL_PIRATE_COSTUME_FEMALE))
+                return false;
+            if (!sSpellMgr->GetSpellInfo(SPELL_NINJA_COSTUME_MALE))
+                return false;
+            if (!sSpellMgr->GetSpellInfo(SPELL_NINJA_COSTUME_FEMALE))
+                return false;
+            if (!sSpellMgr->GetSpellInfo(SPELL_LEPER_GNOME_COSTUME_MALE))
+                return false;
+            if (!sSpellMgr->GetSpellInfo(SPELL_LEPER_GNOME_COSTUME_FEMALE))
+                return false;
+            if (!sSpellMgr->GetSpellInfo(SPELL_GHOST_COSTUME_MALE))
+                return false;
+            if (!sSpellMgr->GetSpellInfo(SPELL_GHOST_COSTUME_FEMALE))
+                return false;
+            if (!sSpellMgr->GetSpellInfo(SPELL_HALLOWED_WAND_PIRATE))
+                return false;
+            if (!sSpellMgr->GetSpellInfo(SPELL_HALLOWED_WAND_NINJA))
+                return false;
+            if (!sSpellMgr->GetSpellInfo(SPELL_HALLOWED_WAND_LEPER_GNOME))
+                return false;
+            if (!sSpellMgr->GetSpellInfo(SPELL_HALLOWED_WAND_RANDOM))
+                return false;
+            if (!sSpellMgr->GetSpellInfo(SPELL_HALLOWED_WAND_SKELETON))
+                return false;
+            if (!sSpellMgr->GetSpellInfo(SPELL_HALLOWED_WAND_WISP))
+                return false;
+            if (!sSpellMgr->GetSpellInfo(SPELL_HALLOWED_WAND_GHOST))
+                return false;
+            if (!sSpellMgr->GetSpellInfo(SPELL_HALLOWED_WAND_BAT))
+                return false;
+            return true;
+        }
+
+        void HandleScriptEffect()
+        {
+            Unit* caster = GetCaster();
+            Unit* target = GetHitUnit();
+
+            if (!caster || !target)
```

Comment:
Testing `void HandleScriptEffect(SpellEffIndex effIndex)` gives this result in VS2013 Update 5:

github.com\tkrokli\TrinityCore\src\server\scripts\Spells\spell_holiday.cpp(348):
```
error C2440: '<function-style-cast>' : cannot convert from 'void (__cdecl spell_hallow_end_wand::spell_hallow_end_wand_SpellScript::* )(SpellEffIndex)' to 'spell_hallow_end_wand::spell_hallow_end_wand_SpellScript::HitHandlerFunction'
              No constructor could take the source type, or constructor overload resolution was ambiguous
...
LINK : fatal error LNK1181: cannot open input file '..\scripts\Release\scripts.lib'
```
Testing `void HandleScriptEffect(SpellEffIndex /*effIndex*/)` gives this result:
```
spell_holiday.cpp(348): error C2039: 'HandleScriptEffect' : is not a member of 'spell_hallow_end_wand::spell_hallow_end_wand_SpellScript'
spell_holiday.cpp(299) : see declaration of 'spell_hallow_end_wand::spell_hallow_end_wand_SpellScript'
error C2065: 'HandleScriptEffect' : undeclared identifier
```
`void HandleScriptEffect()` and removing `if (!caster || !target)` + the next 2 lines => OK.